lib/media/streaming_engine.js

```diff
@@ -1273,11 +1276,15 @@ shaka.media.StreamingEngine = class {
             'ReadableStream is not supported by the browser.');
         }
         const fetchSegment = this.fetch_(mediaState, reference);
-        const result = await fetchSegment;
+        let result = await fetchSegment;
         this.destroyer_.ensureNotDestroyed();
         if (this.fatalError_) {
           return;
         }
+        if (reference.hlsAes128Key && iter) {
```

Comment:
When is the iterator null?  A few lines down on 1310 we assume it's not.

If this is to satisfy the compiler that a nullable thing is non-null, I think an assertion would be preferable (for clarity).  This way, it looks like this is a condition we need to handle by skipping decryption, which doesn't make sense to me.